src/Dfe.PlanTech.Web/Controllers/QuestionsController.cs

```diff
@@ -18,67 +19,68 @@ public class QuestionsController : BaseController<QuestionsController>
     /// </summary>
     /// <param name="id"></param>
     /// <param name="section">Name of current section (if starting new)</param>
-    /// <param name="query"></param>
-    /// <exception cref="ArgumentNullException">Throws exception when Id is null or empty</exception>
     /// <returns></returns>
-    public async Task<IActionResult> GetQuestionById(
-        string id,
-        string? section,
-        int? submissionId,
-        string? answerRef,
-        [FromServices] SubmitAnswerCommand submitAnswerCommand,
-        CancellationToken cancellationToken)
+    public async Task<IActionResult> GetQuestionById(string? id, string? section, [FromServices] SubmitAnswerCommand submitAnswerCommand, CancellationToken cancellationToken)
     {
-        if (string.IsNullOrEmpty(id)) throw new ArgumentNullException(nameof(id));
-        object? parameters;
+        var parameterQuestionPage = TempData[TempDataConstants.Questions] != null ? DeserialiseParameter<TempDataQuestions>(TempData[TempDataConstants.Questions]) : new TempDataQuestions();
 
-        TempData.TryGetValue("param", out parameters);
+        if (string.IsNullOrEmpty(id)) id = parameterQuestionPage.QuestionRef;
 
-        Question? question = null;
+        TempData.TryGetValue("param", out object? parameters);
+        Params? param = _ParseParameters(parameters?.ToString());
 
-        if (submissionId == null)
-        {
-            Params? param = _ParseParameters(parameters?.ToString());
-            var submission = await submitAnswerCommand.GetOngoingSubmission(param?.SectionId ?? throw new NullReferenceException(nameof(param.SectionId)));
-            if (submission != null && !submission.Completed)
-            {
-                question = await submitAnswerCommand.GetNextUnansweredQuestion(submission.Id);
-                if (question == null) return RedirectToAction("CheckAnswersPage", "CheckAnswers", new { submissionId = submission.Id, sectionId = param?.SectionId, sectionName = param?.SectionName });
+        var nextQuestionResult = await submitAnswerCommand.GetNextQuestion(parameterQuestionPage.SubmissionId, id, param?.SectionId ?? throw new NullReferenceException(nameof(param)), section, cancellationToken);
 
-                submissionId = submission.Id;
-            }
+        if (nextQuestionResult.Question == null)
+        {
+            TempData[TempDataConstants.CheckAnswers] = Newtonsoft.Json.JsonConvert.SerializeObject(new TempDataCheckAnswers() { SubmissionId = nextQuestionResult.Submission?.Id ?? throw new NullReferenceException(nameof(nextQuestionResult.Submission)), SectionId = param.SectionId, SectionName = param.SectionName });
```

Comment:
Same here with using `DeserialiseParameter`. Guessing this is intentional but I'm unsure why

Reply:
See above reply, same thingy